nixos/modules/services/network-filesystems/webdav.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.webdav;
+in
+{
+  options = {
+    services.webdav = {
+      enable = mkEnableOption "WebDAV server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.webdav;
+        description = "The WebDAV package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "User account under which WebDAV runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "Group under which WebDAV runs.";
+      };
+
+      settings = mkOption {
+        type = types.anything;
+        default = { };
+        description = "Attrset that is converted and passed as config file.";
+        example = literalExpression ''
+          {
+              address = "0.0.0.0";
+              port = 8080;
+              scope = "/srv/webdav";
```

Comment:
Does nixos even have a /srv by default? Isn't most stuff under /var/lib?

Reply:
There is no consensus or preference on where webdav's data directory should be.  This example is a literal example rather than a default value. I don't mind change the example value to `/var/lib/webdav`, but I don't think `/var/lib/webdav` is a good default option.  The end user is supposed to set this value by themselves.

Reply:
WebDAV is similar to Samba.  We don't set Samba's shared directory to `/var/lib/samba`. In fact, `/var/lib/samba` is occupied by Samba's own data rather than its shared data. The shared directories are usually in `/home`, `/var`, `/srv`.

Reply:
> I don't mind change the example value to /var/lib/webdav, but I don't think /var/lib/webdav is a good default option.

What is a good default then? 

Samba does https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/network-filesystems/samba.nix#L187-L194